src/coreclr/tools/aot/ILCompiler.ReadyToRun/JitInterface/CorInfoImpl.ReadyToRun.cs

```diff
@@ -2257,7 +2249,7 @@ private bool NeedsTypeLayoutCheck(TypeDesc type)
 
         private bool HasLayoutMetadata(TypeDesc type)
```

Comment:
The one place where this method is used looks like dead/unrechable code. 

This method can only return true when `type.IsValueType` is true, but this case is handled in `EncodeFieldBaseOffset` earlier via the `else if (pMT.IsValueType)` check. It means that this method won't ever return true.

Reply:
Nice cleanup, thanks Jan for pointing that out, deleted in 9th commit.